cljfmt/src/cljfmt/core.cljc

```diff
@@ -192,6 +192,28 @@
 (defn pattern? [v]
   (instance? #?(:clj Pattern :cljs js/RegExp) v))
 
+(defn root? [zloc]
+  (nil? (zip/up zloc)))
+
+(defn- top-level-form [zloc]
+  (->> zloc
+       (iterate z/up)
+       (take-while (complement root?))
+       last))
+
+(defn- ns-form? [zloc]
+  (some-> zloc
+          child-sexprs
+          first
+          #{'ns `ns}))
+
+(defn- find-namespace [zloc]
```

Comment:
A more concise way of writing this would be:

```clojure
(defn- top [zloc]
  (if (root? zloc) zloc (recur (z/up zloc))))

(defn- find-namespace [zloc]
  (some-> zloc top (z/find z/next ns-form?) z/sexpr second))
```

Reply:
Done